ESQL: Validate unique plan attribute names (#110488)
elastic/elasticsearch: docs/reference/esql/processing-commands/enrich.asciidoc

The enrich fields from the enrich index that are added to the result as new
columns. If a column with the same name as the enrich field already exists, the
existing column will be replaced by the new column. If not specified, each of
the enrich fields defined in the policy is added.

> If a column has the same name as the enrich field, it will be discarded unless
> that field is given a new name.

Reviewer: ```suggestion
A column with the same name as the enrich field will be discarded unless 
the enrich field is renamed.
```
